src/components/ConfigEditor.tsx

```diff
@@ -95,11 +99,11 @@ export function ConfigEditor(props: DataSourcePluginOptionsEditorProps<FlightSQL
                 width={40}
                 name="password"
                 type="text"
-                value={jsonData.password || ''}
+                value={secureJsonData?.password || ''}
                 placeholder="****************"
                 onChange={(e) => onPasswordChange(e, options, onOptionsChange)}
-                onReset={() => onPasswordChange(null, options, onOptionsChange)}
-                isConfigured={false}
+                onReset={() => onResetPassword(options, onOptionsChange)}
+                isConfigured={secureJsonFields?.password}
```

Comment:
```suggestion
                isConfigured={!!secureJsonFields?.password}
```
Seems like `isConfigured` is a boolean and `password` is a string, adding `!!` making the conversion of a string into a boolean explicit.